### PR TITLE
feat: added websocket-based transmission

### PR DESCRIPTION
**Pull request by Kapim**
added two methods for image and json-data transmission over websockets

code clean-up for middleware communication

### CODE REVIEW

File: examples/client_gstreamer_no_middleware.py [hunk 1]
```diff
@@ -58,11 +58,11 @@ def signal_handler(sig: int, frame: Optional[FrameType]) -> None:
     try:
         # creates the NetApp client with gstreamer extension
         client = NetAppClientGstreamer("", "", "", "", True, get_results, False, False, NETAPP_ADDRESS, NETAPP_PORT)
-        # register the client with the NetApp
-        client.register()
         assert client.netapp_host
         assert client.gstreamer_port
 
+        # register the client with the NetApp
+        client.register()
```
Comment (klepo) [inline examples/client_gstreamer_no_middleware.py:65]: Are you sure about this? client.gstreamer_port is set after registration

File: examples/client_gstreamer.py [hunk 2]
```diff
@@ -65,11 +65,11 @@ def signal_handler(sig: int, frame: Optional[FrameType]) -> None:
         client = NetAppClientGstreamer(
             MIDDLEWARE_ADDRESS, MIDDLEWARE_USER, MIDDLEWARE_PASSWORD, MIDDLEWARE_TASK_ID, True, get_results, True, True
         )
-        # register the client with the NetApp
-        client.register()
         assert client.netapp_host
         assert client.gstreamer_port
 
+        # register the client with the NetApp
```
Comment (klepo) [inline examples/client_gstreamer.py:71]: Are you sure about this? client.gstreamer_port is set after registration

File: examples/client_gstreamer_no_middleware.py (hunk 1 above)
Comment (Kapim) [reply examples/client_gstreamer_no_middleware.py:65]: You are right; I wanted to add the register call after the first assert (netapp_host), which is needed in the register method, but I missed the gstreamer_port assert... I will fix it

Comment (Kapim) [reply examples/client_gstreamer_no_middleware.py:67]: done

File: examples/client_gstreamer.py (hunk 2 above)
Comment (Kapim) [reply examples/client_gstreamer.py:73]: fixed

**Review by klepo — APPROVED** [review]

**Pull request merged**